### DISCUSSION TITLE
How to change the API address used by the dashboard?

### DISCUSSION THREAD

**Discussion by mplucinski** [terminusdb/terminusdb#1459] [Q&A] [answered]
I'm trying to make Terminus API and the dashboard to be available under a different port than the default 6363. I'm using the configuration as below (docker-compose.yml):

```
services:
  database:
    image: "terminusdb/terminusdb-server:latest"
    ports: ["65143:6363"]
```

This makes it possible to access the server under the new port, e.g. I can use curl like: `curl http://localhost:65143/api/` and get an answer.

However, it's not possible to use the dashboard this way. Although opening in the browser `http://localhost:65143/dashboard/admin` loads the page, but then I'm presented with an error:

> Connection Server Error
> 
> API Error url: http://127.0.0.1:6363/api/user_organizations
> [Please contact the TerminusDB Team](https://discuss.terminusdb.com/)

Browser's network tools confirm that it indeed tries to use 127.0.0.1:6363.

My understanding is that the dashbord (web app) is unaware that it should use a different address. How to tell it that? I tried to use environment variables like TERMINUSDB_PORT or TERMINUSDB_SERVER_PORT, but these only change the port the server binds to (in my case inside the container) and seem to have no influence over the served web app.

**Comment by rrooij** [comment, marked as answer]
Good catch! We currently compile the dashboard with an ENV variable to which server it should connect and for the local dashboard, we indeed made this localhost:6363. Unfortunately this is at the moment not changeable at runtime in a very easy way, but we do support overriding the server URL as a workaround if you set the following key/value pair in your browser's localStorage when visiting the dashboard:

```
terminusdb-server-override: http://localhost:65143
```
